staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -218,7 +232,8 @@ func (rules *ClientConfigLoadingRules) Load() (*clientcmdapi.Config, error) {
 	}
 
 	if rules.WarnIfAllMissing && len(missingList) > 0 && len(kubeconfigs) == 0 {
-		klog.Warningf("Config not found: %s", strings.Join(missingList, ", "))
+		missingString := strings.Join(missingList, ", ")
+		rules.Warner.Warn(fmt.Sprintf("Config not found: %s", missingString))
```

Comment:
nit: missingString is only used here, its value could be embedded here directly.